### PR TITLE
Fix mouseMove event type comparison for IE10-11, pointermove

### PR DESCRIPTION
**Pull request by adube**
This PR fixes the issue raised in #3280.  Please, let me know if this fix is robust enough to be accepted / merged, otherwise a review would be welcome and I'll apply any modifications required.

### CODE REVIEW

**Comment by ahocevar** [comment]
What makes me suspicious is that we had mousemove here in the first place, and that pointermove does not work everywhere. cc @tsauerwein

**Comment by tsauerwein** [comment]
`mapBrowserEvent.originalEvent.type` is `pointermove` for IE and `mousemove` for other browsers, that's right. But instead you should check `mapBrowserEvent.type` which should be `pointermove` in all cases.

So, I think we need a condition `ol.events.condition.pointerMove`:

    /**
     * @param {ol.MapBrowserEvent} mapBrowserEvent Map browser event.
     * @return {boolean} True if the browser event is a `pointermove` event.
     * @api
     */
    ol.events.condition.pointerMove = function(mapBrowserEvent) {
      return mapBrowserEvent.type == 'pointermove';
    };

This [line ](https://github.com/openlayers/ol3/blob/master/src/ol/interaction/selectinteraction.js#L193) should probably also be replaced.

**Comment by elemoine** [comment]
+1 for adding a pointerMove condition and use that in the select interaction.

**Comment by adube** [comment]
Just to be clear: should pointerMove replaces mouseMove, or should it be added and mouseMove kept as well ?

**Comment by elemoine** [comment]
`mouseMove` does not have the `stable` marker, so I think we can just replace `mouseMove` by `pointerMove`.

**Comment by adube** [comment]
@elemoine, got it.  I'll update the PR right away.

**Comment by adube** [comment]
There your go.

**Comment by elemoine** [comment]
LGTM. @tsauerwein, please merge if you agree with the patch as well.

**Comment by ahocevar** [comment]
Thanks everyone. To me, everything looks consistent now.

**Comment by fredj** [comment]
`ol.events.condition.mouseMove` is an api function; shouldn't we keep it?

**Comment by elemoine** [comment]
> ol.events.condition.mouseMove is an api function; shouldn't we keep it?

It's not marked as stable, so we're free to do what we want with it, including replacing it with something else.

**Comment by tsauerwein** [comment]
> It's not marked as stable, so we're free to do what we want with it, including replacing it with something else.

Yes it's not marked as stable, but all the users that used `ol.events.condition.mouseMove` like in the example will have a problem with the next version. I would simply keep `ol.events.condition.mouseMove`. If not we should mention this in the upgrade notes.

Besides, the changes look good to me. Thanks, @adube!

**Comment by ahocevar** [comment]
My opinion: Let's remove `ol.events.condition.mouseMove` and fully commit to the pointer events API. All it takes is a mention in the release notes.

**Comment by elemoine** [comment]
> My opinion: Let's remove ol.events.condition.mouseMove and fully commit to the pointer events API. All it takes is a mention in the release notes.

+1. I was going to suggest the same.

**Comment by tsauerwein** [comment]
That also sounds good to me.

**Comment by tschaub** [comment]
Makes sense to me as well.  It would be nice to have a note for the release accompany this change.  We could add a `changelog/v3.x.md` file that included the upgrade notes.  When the next release is cut, the release manager would move all upgrade notes into the appropriate release changelog (e.g. `changelog/v3.3.0.md`).

@adube would you be able to add a `changelog/v3.x.md` file with upgrade notes about this change?

**Comment by elemoine** [comment]
> @adube would you be able to add a changelog/v3.x.md file with upgrade notes about this change?

+1 !!

**Comment by gberaudo** [comment]
The issue with moving the information from one file to another is that
history gets lost.

On the contrary, if the information is wrote during the commit that bring
the change and never moved around, then `git blame` will work.

For expamle, this workflow would work:

- commit A (by the commiter)
Change something.
Prepend a note to the CHANGES.md filep

- commit B (by the commiter)
Change something.
Prepend a note to the CHANGES.md filep

...

- Release version v3.3.0 (by the release manager)
Prepend the release numbers / date to the CHANGES.md filep

CHANGES.md is an example name.


Using `git blame` on the CHANGES.md file shows the list of all the commits
having brought a change. A git show on a particular commit leads to the
exact changes, PR and discussion.

**Comment by adube** [comment]
I understand your point, @gberaudo.

How about we have a CHANGES.md, which would contain all the release notes of all releases, and the developer making changes required to be noted would be made there.

Then, upon creating a new release, the release manager copies the top-most section and create the new 3.3.0.md file from it.

@tschaub, @elemoine, I'll do as you asked for now.  Just le me know if you want me to do otherwise.

**Comment by elemoine** [comment]
@gberaudo what do you suggest? Using one CHANGES.md file and include all the release notes and changes in this file? This file would quickly become huge and inconvenient.

**Comment by adube** [comment]
@elemoine, how about having both ?  (one huge file for the devs to edit, and 1 file per release for the release manager to create)

**Comment by elemoine** [comment]
> @elemoine, how about having both ? (one huge file for the devs to edit, and 1 file per release for the release manager to create)

Ok for me.

**Comment by adube** [comment]
Gotcha.  I won't include all of the past release notes as part of this PR.  I'll create an other PR for that.

**Comment by adube** [comment]
Done.  Please, let me know if everything's fine.

File: changelog/CHANGES.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+## Summary
+
+## Upgrade notes
+
+The following changes were made to the API:
+
+ * `ol.events.condition.mouseMove` was replaced by `ol.events.condition.pointerMove`, so your `ol.interaction.Select` that were using it have to be updated. (see [#3281](https://github.com/openlayers/ol3/pull/3281)).
+
+
+## Changes
```
Comment (elemoine) [inline changelog/CHANGES.md]: I don't think we need the `Changes` section. This one is automatically generated.

**Comment by adube** [comment]
Ready for review.

**Comment by gberaudo** [comment]
@elemoine, one huge CHANGES.md file is one possibility to get easy to
`git blame` changelog. Another solution could be to use one file per
month will also do. It would look like:

changelog/v3.4.x
- Release 3.4.0 (by the release manager)
Some changes (by the commiters)

changelog/v3.3.x
- Release 3.3.1 (by the release manager)
One change (by the commiter)
Another change (by the commiter)
- Release 3.3.0 (by the release manager)
Some changes (by the commiters)

Do you prefer this way?

**Comment by adube** [comment]
@elemoine, will this be merged before 3.3.0 ?

**Comment by elemoine** [comment]
@adube, could you just please remove the CHANGES.md file? I'll merge your branch with this file and add it later by myself. Thanks.

**Comment by adube** [comment]
There you go !

**Comment by adube** [comment]
Thank you very much for all of you who participated.

**Comment by elemoine** [comment]
I've realized about a problem this change introduced on touch devices. `pointermove` events are triggered while dragging/panning the map on touch devices. This means that panning the map will select features on touch devices.

To fix the problem we could introduce a new event condition function, `ol.events.condition.pointerMoveNoDragging`, and use that function instead of `ol.events.condition.pointerMove` with the Select interaction. The function would look like this:

```js
/**
 * @param {ol.MapBrowserEvent} mapBrowserEvent Map browser event.
 * @return {boolean} True if the browser event is a `pointermove` event.
 * @api
 */
ol.events.condition.pointerMoveNoDragging = function(mapBrowserEvent) {
  return ol.events.condition.pointerMove(mapBrowserEvent) &&
      mapBrowserEvent.dragging == false;
};
```

**Pull request merged**